config/config.go

```diff
@@ -508,6 +522,49 @@ func (c *ServersetSDConfig) UnmarshalYAML(unmarshal func(interface{}) error) err
 	return checkOverflow(c.XXX, "serverset_sd_config")
 }
 
+// KubernetesSDConfig is the configuration for Kubernetes service discovery.
+type KubernetesSDConfig struct {
+	Server           string   `yaml:"server"`
+	NodeMetricsPort  int      `yaml:"node_metrics_port,omitempty"`
+	InCluster        bool     `yaml:"in_cluster,omitempty"`
+	BearerTokenFile  string   `yaml:"bearer_token_file,omitempty"`
+	Username         string   `yaml:"username,omitempty"`
+	Password         string   `yaml:"password,omitempty"`
+	Insecure         bool     `yaml:"insecure,omitempty"`
+	CertFile         string   `yaml:"cert_file,omitempty"`
+	KeyFile          string   `yaml:"key_file,omitempty"`
+	CAFile           string   `yaml:"ca_file,omitempty"`
+	RetryDelay       Duration `yaml:"retry_delay,omitempty"`
+	RequestTimeout   Duration `yaml:"request_timeout,omitempty"`
+	DiscoverServices bool     `yaml:"discover_services,omitempty"`
```

Comment:
The problem is that not all services will be scrapeable. This discovery requires that services are annotated with configurable annotations to enable scraping. Enabling this for everything by default will only increase noise in a heterogeneous env IMO.

This flag is actually there to disable service scraping totally so you only scrape nodes & get container (crgoup) metrics, rather than app level metrics.